### PR TITLE
Improve tracing of Fault Tolerance

### LINKED ISSUE #6526: FT 2.0: Trace reason for not retrying
Currently we have fairly good trace for most events which occur during fault tolerance processing, including whether or not we retry. However, we don't currently trace the reason why we didn't retry, which would have been useful to me recently.

### LINKED ISSUE #6664: FT 2.0: Use better Ids for tracing calls
In our trace messages, we currently use the fully qualified method name to identify the execution which is taking place.

This has two downsides:
1) The fully qualified method name is very long
2) The trace is very confusing when the same method is being run multiple times concurrently

It might be better to assign to random identifier at the start of each invocation, have the first trace message include both the ID and the fully qualified method name, and then just use the identifier in future trace messages.

### PR DESCRIPTION
**Pull request by Azquelt**
Generates a random ID for each execution, resulting in shorter messages
which are easier to track when there are multiple executions running
concurrently.

For FT 1.x, change from using a UUID to using a shorter pseudorandom
string

Also trace out the reason why when a retry does not occur.

Fixes #6664 
Fixes #6526

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

File: dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/impl/AsyncExecutor.java [hunk 1]
```diff
@@ -246,14 +246,14 @@ private void runExecutionAttempt(AsyncAttemptContextImpl<W> attemptContext, Bulk
             }
 
             if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
-                Tr.event(tc, "Method {0} attempt execution reuslt: {1}", executionContext.getMethod(), methodResult);
+                Tr.event(tc, "Execution {0} attempt execution reuslt: {1}", executionContext.getId(), methodResult);
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/impl/AsyncExecutor.java]: "reuslt"

Comment (tevans78) [reply dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/impl/AsyncExecutor.java]: Also, do you need to say "execution" twice? Maybe just "Execution {0} result: {1}"?

File: dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/RetryStateImpl.java [hunk 2]
```diff
@@ -55,31 +55,31 @@ public RetryResult recordResult(MethodResult<?> methodResult) {
...
             }
         } else {
             // Successful case
-            result.shouldRetry = false;
+            result.reason = RetryResultReason.NO_EXCEPTION;
         }
 
         // Capture whether this result was considered a failure by the Retry
-        boolean resultWasFailure = result.shouldRetry;
+        boolean resultWasFailure = result.shouldRetry();
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/RetryStateImpl.java]: you're asking the result if it thinks it should retry and then inferring that the result was a failure. Perhaps the method should be `result.wasFailure()`?

File: dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/RetryStateImpl.java [hunk 3]
```diff
@@ -171,6 +171,31 @@ public long getDelay() {
         public TimeUnit getDelayUnit() {
             return delayUnit;
         }
+
+        @Override
+        public String toString() {
+            return reason.toString();
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/RetryStateImpl.java]: `reason` is not guaranteed to be non-null, you can't use it like this in a toString

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceInterceptor.java [hunk 4]
```diff
@@ -242,6 +244,15 @@ private Object execute(InvocationContext invocationContext, AggregatedFTPolicy a
         return result;
     }
 
+    private String generateId(Method method) {
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceInterceptor.java]: `@Trivial`?

**Comment by Azquelt** [comment]
squashed review changes

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Last build failed with an infrastructure communication failure.

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Failures in the build do not appear to be related to this change.

**Pull request merged**